docs/docs/jsonnet/native-functions.md

```diff
@@ -6,65 +6,216 @@ menu: "Writing Jsonnet"
 
 # Native Functions
 
-Tanka extends the Jsonnet standard library with some common helper utilities.
+Native functions are Jsonnet functions implemented in Go rather than Jsonnet itself. Tanka extends the Jsonnet native functions with some common helper utilities.
 
-## parseJSON
+To use a native function in your Jsonnet code, you need to use the `native` function from the Jsonnet standard library. For example, to parse a Kubernetes deployment YAML file into a Jsonnet object:
 
-```ts
-parseJSON(string json) Object
+```jsonnet
+{
+  deployment:  std.native('parseYaml')(importstr 'deployment.yaml'),
+}
 ```
 
-`parseJSON` wraps `json.Unmarshal` to convert a json string into a dict
+The argument to `std.native` is the name of the native function to lookup. In this case, it is `parseYaml`. The returned function is then evaluated with the arguments in the second set of parentheses. This is the `deployment.yaml` file imported as a string using the Jsonnet keyword `importstr`.
 
-## parseYAML
+## parseJson
 
-```ts
-parseYAML(string yaml) []Object
+### Usage
+
+```jsonnet
+std.native('parseJson')(json)
+```
+
+`parseJson` parses a JSON string and returns the JSON element.
+
+### Examples
+
+```jsonnet
+{
+  parseJsonArray: std.native('parseJson')('[0, 1, 2]'),
+  parseJsonObject: std.native('parseJson')('{ "foo": "bar" }'),
+}
+```
+
+Evaluating with Tanka results in the JSON:
+
+```json
+{
+  "parseJsonArray": [0, 1, 2],
+  "parseJsonObject": {
+    "foo": "bar"
+  }
+}
+```
+
+## parseYaml
+
+### Usage
+
+```jsonnet
+std.native('parseYaml')(yaml)
+```
+
+`parseYaml` parses a string of YAML document(s) and returns an array of JSON objects. If `yaml` contains only a single document, a single value array will be returned.
+
+### Examples
+
+```jsonnet
+{
+  parseYaml: std.native('parseYaml')(|||
```

Comment:
Might be confusing to have the nativeFuncs name as the object key here – which is obviously not a requirement. 

Maybe rename all keys to something describing the content, such as `yaml`, `escaped`, `matched`, etc?

Reply:
That's true, and your suggestion for the renames will make that a lot clearer and understandable for sure.